### PR TITLE
Feature/gallery domain service

### PR DESCRIPTION
**Pull request by AdrianTan166**
## Gallery Domain + Service layer

Awaiting backend feature completion to fully test

## Checklist

- [x] I have reviewed my own code
- [x] I have verified that my code created matches the **development strategy**
- [x] I have verified that all Acceptance Criteria (AC) are met
- [ ] I have run this on a physical device
- [x] I have verified that my commits are properly labelled
- [x] I have updated any related documentation
- [x] I have added appropriate tests for my new changes to at least 80% coverage
- [x] I have verified that code passes the CI/CD review
- [x] I have updated any existing tests related to my changes

### CODE REVIEW

**Comment by karnull** [comment]
haven't ran it on my own machine (mainly cause im lacking the env) but going through gitlogs the work here seems fine. I say that while acknowledging my rather newbie webdev brain. onto @Darragh-Hazell to tick it off. Also @RolandWarburton if you got any feedback

**Comment by RolandWarburton** [comment]
Been a busy week, ill check it tmr :)

I also checked the diff and it looks nice so far

**Comment by RolandWarburton** [comment]
Potentially missed pokeapi reference in cypress.config.ts:22:30

are we planning on creating tests, or is it a "as time permits" kind of situation?

File: env/.env.local [hunk 1]
```diff
@@ -1,5 +1,5 @@
 REACT_APP_STAGE=LOCAL
 REACT_APP_VERSION=GITHUB_RUN_NUMBER
-REACT_APP_CHALLENGE_BASE_URL=https://pokeapi.co/api/v2
+REACT_APP_CHALLENGE_BASE_URL=https://astrostreakstest.co/api
```
Comment (RolandWarburton) [inline env/.env.local:3]: if this is running locally on the machine it may be localhost and not the domain name?

**Comment by RolandWarburton** [comment]
Why is the folder `src/components/pages/MovieInfo` called that? should it be changed to `ImageInfo`?

I think this because the page contained inside `src/components/pages/MovieInfo/ImageInfo.tsx` is used on the `/image/:id` route which will render an image.

**Comment by RolandWarburton** [comment]
`infra/` Is not being used anywhere, i don't think we are planning on deploying to AWS so can be removed i believe (tests still pass if deleted).

**Comment by karnull** [comment]
> Potentially missed pokeapi reference in cypress.config.ts:22:30
> 
> are we planning on creating tests, or is it a "as time permits" kind of situation?

will potentially write some tests

**Comment by RolandWarburton** [comment]
after reviewing my comments can go ahead and merge, looks like you got most of the boilerplate appropriated for astrostreak :shipit:

File: env/.env.local (hunk 1 above)
Comment (AdrianTan166) [reply env/.env.local:3]: Yep, once backend is running, we can switch this to localhost

**Comment by AdrianTan166** [comment]
> Potentially missed pokeapi reference in cypress.config.ts:22:30
> 
> are we planning on creating tests, or is it a "as time permits" kind of situation?

Ahh yep will fix the reference in Cypress. Unit tests we will right, UI tests will be a 'as time permits'

**Comment by AdrianTan166** [comment]
> Why is the folder `src/components/pages/MovieInfo` called that? should it be changed to `ImageInfo`?
> 
> I think this because the page contained inside `src/components/pages/MovieInfo/ImageInfo.tsx` is used on the `/image/:id` route which will render an image.

Ahh yep missed refactor, thanks

**Comment by karnull** [comment]
> after reviewing my comments can go ahead and merge, looks like you got most of the boilerplate appropriated for astrostreak :shipit:

dang i didn't know about emotes in github community

**Review by RolandWarburton — APPROVED** [review]

**Review by RolandWarburton — APPROVED** [review]

**Review by RolandWarburton — APPROVED** [review]
approved

**Review by karnull — APPROVED** [review]
aight good enough to merge imo

**Review by RolandWarburton — APPROVED** [review]

**Review by RolandWarburton — APPROVED** [review]

**Review by karnull — APPROVED** [review]

**Pull request merged**